src/library-authoring/create-unit/CreateUnitModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import {
+  ActionRow,
+  Button,
+  Form,
+  ModalDialog,
+} from '@openedx/paragon';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import FormikControl from '../../generic/FormikControl';
+import { useLibraryContext } from '../common/context/LibraryContext';
+import messages from './messages';
+import { useCreateLibraryContainer } from '../data/apiHooks';
+import { ToastContext } from '../../generic/toast-context';
+
+const CreateUnitModal = () => {
+  const intl = useIntl();
+  const {
+    libraryId,
+    isCreateUnitModalOpen,
+    closeCreateUnitModal,
+  } = useLibraryContext();
+  const create = useCreateLibraryContainer(libraryId);
+  const { showToast } = React.useContext(ToastContext);
+
+  const handleCreate = React.useCallback((values) => {
+    create
+      .mutateAsync({
+        containerType: 'unit',
+        ...values,
+      })
+      .then(() => {
+        closeCreateUnitModal();
+        // TODO: Navigate to the new unit
+        // navigate(`/library/${libraryId}/units/${data.key}`);
+        showToast(intl.formatMessage(messages.createUnitSuccess));
+      })
+      .catch(() => {
+        showToast(intl.formatMessage(messages.createUnitError));
+      });
+  }, []);
+
+  return (
+    <ModalDialog
+      title={intl.formatMessage(messages.createUnitModalTitle)}
+      isOpen={isCreateUnitModalOpen}
+      onClose={closeCreateUnitModal}
+      hasCloseButton
+      isFullscreenOnMobile
```

Comment:
```suggestion
      isFullscreenOnMobile
      isOverflowVisible={false}
```

Just to be safe from this [issue](https://github.com/openedx/frontend-app-authoring/pull/1699#discussion_r1987054071).